tests/problem4.test.js

```diff
@@ -1,29 +1,47 @@
-/* 
+/*
     Napisz funkcję takeFirst(count, ...promises).
 
-    Ma ona zwrócić Promise z tablicą wyników ${count} najszybciej rozwiązanych Promisów. 
+    Ma ona zwrócić Promise z tablicą wyników ${count} najszybciej rozwiązanych Promisów.
     Jeśli żaden Promise nie zostanie podany, wynikiem powinna być pusta tablica.
-    W przypadku błędu któregoś z nich (jeśli wymagana liczba Promisów jeszcze się nie rozwiązała), 
+    W przypadku błędu któregoś z nich (jeśli wymagana liczba Promisów jeszcze się nie rozwiązała),
     to powinna zwrócić odrzucony Promise z odpowiednim błędem.
-    
+
     Przykład:
 
     const p1 = p.resolveWith(5).after(5) // resolves in 5 sec.
     const p2 = p.resolveWith(3).after(3) // resolves in 3 sec.
     const p3 = p.resolveWith(2).after(2) // resolves in 2 sec.
     const p4 = p.resolveWith(6).after(6) // resolves in 6 sec.
-    const r1 = p.rejectWith(4).after(4) // rejects in 4 sec. 
+    const r1 = p.rejectWith(4).after(4) // rejects in 4 sec.
 
-    takeFirst(2, p1, p2, p3, p4, r1).then(console.log) // [p3, p2, p1]
+    takeFirst(2, p1, p2, p3, p4, r1).then(console.log) // [p3, p2]
 
     takeFirst(2, p1, p4, r1).catch(console.log) // error thrown by r1
 
 */
 
 const p = require('../utils.js');
 
+async function takeFirst(count, ...promises) {
+    let result = [];
+    if (!promises || promises.length == 0)
+        return result;
+
+    const pushed = promises.map(prom => prom.then(function(value) {
```

Comment:
ponieważ ten `push` to wyraźny efekt uboczny, używanie `map` nie jest tu wskazane. Równie dobrze sprawdziłby się `forEach`. `map` komunikuje, że wykonujesz "czyste" obliczenie na poszczególnych elementach tablicy, a tutaj to nieprawda.